### PR TITLE
New intermediate images

### PR DESCRIPTION
**Pull request by ericof**

### CODE REVIEW

File: Dockerfile.builder
Comment (fredvd) [inline Dockerfile.builder:2]: Add comment for viewer:
This is the base image that installs the build environment and requirements. You depend on this builder image in customised images for dev, production, acceptance or your own projects.

File: Dockerfile.dev
Comment (fredvd) [inline Dockerfile.dev:2]: Add comment:

This image is depending on the builder image to serve as a development environment for either the Plone 'content api' backend for Volto or the Plone classicUI server

File: Dockerfile.prod
Comment (fredvd) [inline Dockerfile.prod:2]: We have a small issue here:  .prod is too generic because this is .prod for the Volto frontend, but we need to name a second Dockerfilehere that generates the prod-classicui server image.

Add comment:

This is a basic production image for the Plone 'content api' backend that is required for the react based  plone-frontend  image (Volto)

**Review by fredvd — CHANGES_REQUESTED** [review]
See remark on .prod.  We need to conside naming if we also generate plone/server-classicui here.

**Comment by jensens** [comment]
IMO it is a good idea to document the dependency chain of the images created, with its purpose. Together with Frontend.

**Comment by ericof** [comment]
@jensens It is my to-do list. I plan to use [JupyterLab](https://jupyter-docker-stacks.readthedocs.io/en/latest/using/selecting.html#image-relationships) as an example

**Comment by stevepiercy** [comment]
@ericof in Plone 6 Documentation, we use [Graphviz](https://6.docs.plone.org/contributing/setup-build.html#graphviz) for graphic visualization. See examples:

- https://github.com/plone/documentation/search?q=graphviz
- https://deploy-preview-1442--6-docs-plone-org.netlify.app/classic-ui/views.html (the current build is broken, but the preview works fine)

**Review by fredvd — APPROVED** [review]
LGTM, but lets test after merging,

**Pull request merged**
